### PR TITLE
udev: slightly extend comment and add more tests

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for #26693.

### CODE REVIEW

File: test/units/testsuite-50.sh [hunk 1]
```diff
@@ -447,6 +447,19 @@ fi
...
 
+# Test long reference name.
+# Note, sizeof_field(struct loop_info64, lo_file_name) == 64,
+# hence we can set 63 char with trailing NUL.
+name="$(for ((i = 0; i < 63; i++)); do echo -n x; done)"
+LOOP="$(systemd-dissect --attach --loop-ref="$name" "${image}.raw")"
+udevadm trigger -w "$LOOP"
+
+# Check if the /dev/loop/* symlinks really reference the right device
+test "/dev/loop/by-ref/$name" -ef "$LOOP"
```
Comment (yuwata) [inline test/units/testsuite-50.sh:459]: Note, without the first commit, this check fails, although `systemd-dissect --attach --loop-ref=$name` in the above succeeds.

**Comment by poettering** [comment]
i don't follow. So let's say i attach one loopback block device and set its lo_file_name fields to 64 'x' characters (and no trailing NUL byte). and I attach another one to 63 'x' characters (and one trailing NUL byte). in both cases the ioctl() to configure the device will happily take what we pass. One gets truncated though, the other one does not.

When i read the string back, from the blkid udev builtin, via the ioctl, both will look exactly the same, i cannot correctly determine truncation of the data returned by the kernel, both fields will have 63 'x' bytes and one NUL byte in it.

My code so far then said "ok, i simply don't know if it got truncated or not. But since the goal of detecting truncation is to avoid ambiguity i'll simply step away and assume the worst".

this is different from the case where let's say 62 'x' bytes were set as label. i read that back and there's no ambiguity: it's definitely not truncated.

**Comment by yuwata** [comment]
Oh, I did not know that the kernel automatically truncates the input string (done by `loop_set_status_from_info()`)...

**Comment by yuwata** [comment]
@poettering Sorry for the noise and thank you for the explanation.
The root of my confusion is that I did not know the kernel truncates the name, I thought the kernel uses the string as is.
I explicitly mention that the kernel truncates the name.
PTAL.

**Pull request merged**